### PR TITLE
Add notes on client guidelines and link to docs.

### PR DESCRIPTION
**Pull request by michaelkaye**

### CODE REVIEW

**Comment by michaelkaye** [comment]
@MidhunSureshR @bwindels would be interested in thoughts/feedback on what i've expressed here so far on naming etc.

**Comment by michaelkaye** [comment]
Fixes #44

File: docs/client-guidelines.md [hunk 1]
```diff
@@ -1,11 +1,89 @@
...
+
+We Should Not name actions passively, or indirectly
+ * `invite_user` instead of `do_the_user_invite`
+ * `crosssign_user` instead of `invoke_crosssign`
+
+We Should name actions after what they do, not what test they're related to
+ * `create_poll` not `pr_52_action` or `msc_1235_action`
+
+We Should avoid naming actions in a client specific way
+ * `clear_idb_cache` is not great because it's only available in element web
```
Comment (MidhunSureshR) [inline docs/client-guidelines.md:39]: This isn't client specific but rather platform specific. I imagine similar actions would be needed for other platforms as well.

**Review by MidhunSureshR — COMMENTED** [review]
LGTM

File: docs/client-guidelines.md (hunk 1 above)
Comment (michaelkaye) [reply docs/client-guidelines.md]: Have added some possibly clearer negative examples around mouse hovering etc.

**Pull request merged**